### ISSUE TITLE
Brew on Yosemite now installs  0.14-DEV

### ISSUE THREAD

**Issue by tnolet** [spf13/hugo#918]
This kinda had me pulling my hair out for 2 hours. Then I downloaded 0.12 and everything was fine.
Maybe change this.

**Comment by quoha** [comment]
I looked at the brew formula for Hugo. It's grabbing the v0.12.tag.gz tarball from the archive:

```
class Hugo < Formula
  homepage "http://hugo.spf13.com/"
  head "https://github.com/spf13/hugo.git"
  url "https://github.com/spf13/hugo/archive/v0.12.tar.gz"
  sha1 "f0537942cde9645ee2d98aaaf927a80c79070e99"

  bottle do
    revision 1
    sha1 "e892d9a0fb7c832fb7f24db8682d4775dfbcfd04" => :yosemite
    sha1 "42ad7067cabf84b2015e8086eb84115fce3c0e58" => :mavericks
    sha1 "f9c04d315507921b303aed2693438cbc9ef18ca1" => :mountain_lion
  end

  depends_on "go" => :build
  depends_on "bazaar" => :build
  depends_on :hg => :build

  def install
    ENV["GOBIN"] = bin
    ENV["GOPATH"] = buildpath
    ENV["GOHOME"] = buildpath
    system "go", "get"
    system "go", "build", "main.go"
    bin.install "main" => "hugo"
  end

  test do
    site = testpath/"hops-yeast-malt-water"
    system "#{bin}/hugo", "new", "site", site
    assert File.exist?("#{site}/config.toml")
  end
end
```

The `commands/version.go` in that tarball does say v0.12.

```
var version = &cobra.Command{
	Use:   "version",
	Short: "Print the version number of Hugo",
	Long:  `All software has versions. This is Hugo's`,
	Run: func(cmd *cobra.Command, args []string) {
		fmt.Println("Hugo Static Site Generator v0.12")
	},
}
```

And brew does think that it has installed v0.12. The output of `hugo-0.12 version` is the v0.13 tag:

```
$ ll /usr/local/Cellar/hugo/0.12/bin
total 56176
drwxr-xr-x  4 quoha  staff       136 Jan 29 16:25 .
drwxr-xr-x  6 quoha  staff       204 Jan 29 16:25 ..
-r-xr-xr-x  1 quoha  wheel  14381052 Jan 29 16:25 hugo
-r-xr-xr-x  1 quoha  staff  14381052 Jan 29 16:25 hugo-0.12
$ which hugo-0.12
/usr/local/bin/hugo-0.12
$ hugo-0.12 version
Hugo Static Site Generator v0.13-DEV BuildDate: 2015-01-29T16:25:07-06:00
```

Obviously, this won't be a big issue after the v0.13 release, but it is confusing to users and should be addressed.

**Comment by prakhar1989** [comment]
@quoha Sorry if this has been answered before - is there an expected timeline for release of the 0.13 version? Can't wait to get better support for Pygments. As always, thanks to you & the rest of the contributors for their hard work!

**Comment by anthonyfok** [comment]
@prakhar1989: Good news!  v0.13 is released 3 hours ago!  Many thanks to @spf13 and all 65 wonderful contributors!

**Comment by prakhar1989** [comment]
@anthonyfok That is great news! I guess, v0.13 is not available on brew yet?

```shell
$ brew upgrade hugo
Error: hugo-0.12 already installed
```

**Comment by k4rtik** [comment]
Someone has sent a PR already: https://github.com/Homebrew/homebrew/pull/37050

We might need to fix the output of `hugo version` though (says 0.14-dev), if it's such a huge issue.

**Comment by dunn** [comment]
The reason for this that `main.go` requests the HEAD of `github.com/spf13/hugo/commands`, which includes the version command, right?  Can it be pinned to a specific tag or revision instead?

**Comment by bep** [comment]
If it says 0.14-dev it's not 0.13.

It should build from this tag:

https://github.com/spf13/hugo/releases/tag/v0.13

**Comment by mdhender** [comment]
I think that it is a big issue because it's one of the first thing that new users see. Should make that first impression a quality impression.

**Comment by bep** [comment]
The above suggests that the Brew build process is broken for Go. My best guess is that this is a path issue, and that they are are pulling in the latest masters via go get.

If hugo version says 0.14-DEV it IS 0.14-DEV and not 0.13.

**Comment by spf13** [comment]
@bep is correct. Brew should install the stable tag 0.13 by default. 

Steve Francia
spf13.com
@spf13

**Comment by spf13** [comment]
Oh. I see how things could be wrong as even the stable tag could include other libraries from their head. Why not just have brew install from the binary by default?

**Comment by tnolet** [comment]
I can confirm also that behaviour of Hugo was different with the brew install, i.e. I had some bugs that vanished once I installed the 0.12 binary. So I guess it was not a case of the 0.12 brew version "wrongly" reporting 0.13-dev. I was actually running 0.13-dev.

**Comment by mdhender** [comment]
The build log shows that it is doing a `go get`...
```
==> Downloading https://github.com/spf13/hugo/archive/v0.12.tar.gz
Already downloaded: /Library/Caches/Homebrew/hugo-0.12.tar.gz
==> go get
==> go build main.go
🍺  /usr/local/Cellar/hugo/0.12: 5 files, 27M, built in 45 seconds
```

**Comment by mdhender** [comment]
I just opened https://github.com/Homebrew/homebrew/issues/37083.

**Comment by mdhender** [comment]
Curious. Maybe the homebrew team did the build from HEAD this morning?

```
$ wget https://homebrew.bintray.com/bottles/hugo-0.13.yosemite.bottle.tar.gz
$ ll
total 15200
drwxr-xr-x  3 quoha  wheel      102 Feb 22 14:42 .
drwxrwxrwt  8 root   wheel      272 Feb 22 14:41 ..
-rw-r-----@ 1 quoha  staff  7781346 Feb 22 14:42 hugo-0.13.yosemite.bottle.tar.gz
$ tar zxvf hugo-0.13.yosemite.bottle.tar.gz 
x hugo/0.13/
x hugo/0.13/bin/
x hugo/0.13/INSTALL_RECEIPT.json
x hugo/0.13/LICENSE.md
x hugo/0.13/README.md
x hugo/0.13/bin/hugo
x hugo/0.13/bin/hugo-0.13
$ cd hugo/0.13/
$ ll
total 32
drwxr-xr-x@ 6 quoha  wheel   204 Feb 22 02:09 .
drwxr-xr-x  3 quoha  wheel   102 Feb 22 14:42 ..
-rw-r--r--@ 1 quoha  wheel   279 Feb 22 02:09 INSTALL_RECEIPT.json
-rw-r--r--@ 1 quoha  wheel  2707 Feb 21 21:31 LICENSE.md
-rw-r--r--@ 1 quoha  wheel  6748 Feb 21 21:31 README.md
drwxr-xr-x@ 4 quoha  wheel   136 Feb 22 02:09 bin
$ cat INSTALL_RECEIPT.json 
{"used_options":[],"unused_options":[],"built_as_bottle":true,"poured_from_bottle":false,"tapped_from":"Homebrew/homebrew","time":1424592569,"HEAD":"fb6123b42b59ccff0547a5fc1b10bff9427efcbb","stdlib":null,"compiler":"clang","source":{"path":"/usr/local/Library/Formula/hugo.rb"}}Michaels-Mac-Pro:0.13 quoha$ 
$ cd bin
$ ll
total 55280
drwxr-xr-x@ 4 quoha  wheel       136 Feb 22 02:09 .
drwxr-xr-x@ 6 quoha  wheel       204 Feb 22 02:09 ..
-r-xr-xr-x@ 1 quoha  wheel  14150036 Feb 22 02:09 hugo
-r-xr-xr-x@ 1 quoha  wheel  14150036 Feb 22 02:09 hugo-0.13
$ ./hugo version
Hugo Static Site Generator v0.14-DEV BuildDate: 2015-02-22T02:09:28-06:00
```

**Comment by madranet** [comment]
This has just caught me out again, after I stupidly did a "brew upgrade" the other day and wondered why the hell 'hugo server'  was suddenly throwing up a load of parsing errors on a site which had previously been working perfectly. A quick fix [at least on my setup] is to do:

````
rm /usr/local/bin/hugo
ln -s /usr/local/Cellar/hugo/0.12/bin/hugo-0.12 /usr/local/bin/hugo
````
which relinks the 'hugo' command alias back to the previous [working] version

*[I know that probably doesn't help other people much, but at least it gives me a chance of remembering what I did, next time it goest tits-up!]*

**Comment by mdhender** [comment]
@madranet If it's really an issue, install 0.12 from the tarball into your ~/bin folder. Then you have complete control over the upgrades. Plus you won't get that oddball version string mismatch.

**Comment by dunn** [comment]
The Homebrew maintainers [prefer](https://github.com/Homebrew/homebrew/issues/37083#issuecomment-75525203) that dependencies are specified in the formula and fetched by Homebrew rather than by `go get`, but Hugo has a lot of dependencies and they have their own dependencies [and this seems to do the trick anyway](https://github.com/dunn/homebrew/commit/b3bbf48c08db93a36fb61c97c50b01bccaa55901):

```diff
diff --git a/Library/Formula/hugo.rb b/Library/Formula/hugo.rb
index c9e8ba5..f972309 100644
--- a/Library/Formula/hugo.rb
+++ b/Library/Formula/hugo.rb
@@ -1,8 +1,16 @@
 class Hugo < Formula
   homepage "http://gohugo.io/"
   head "https://github.com/spf13/hugo.git"
-  url "https://github.com/spf13/hugo/archive/v0.13.tar.gz"
-  sha1 "a821fcde92b03baf49a45970d0ae6b781a3b12c1"
+
+  stable do
+    url "https://github.com/spf13/hugo/archive/v0.13.tar.gz"
+    sha1 "a821fcde92b03baf49a45970d0ae6b781a3b12c1"
+    # corrects repository url for osext; remove in next release
+    patch do
+      url "https://github.com/spf13/hugo/commit/967d001ebe40cfe90992d953880356a495216202.diff"
+      sha256 "5a991372a54ac049783926c1993d920f46ab52563ff1877e7aaddc40b5a79e22"
+    end
+  end
 
   bottle do
     cellar :any
@@ -19,6 +27,10 @@ def install
     ENV["GOBIN"] = bin
     ENV["GOPATH"] = buildpath
     ENV["GOHOME"] = buildpath
+
+    mkdir_p buildpath/"src/github.com/spf13/"
+    ln_sf buildpath, buildpath/"src/github.com/spf13/hugo"
+
     system "go", "get"
     system "go", "build", "main.go"
     bin.install "main" => "hugo"
```

If I got that right, let me know and I'll open a PR over there.

**Comment by dunn** [comment]
Fixed in Homebrew/homebrew@e97ce3e.

**Comment by anthonyfok** [comment]
Cool!  Thanks for everybody involved in getting this problem resolved!  :-)
